forem/forem
# Member index: Filter by status

## What type of PR is this? (check all applicable)


## Description

This adds the ability to filter by "status" to the member index view. "Statuses" are just a subset of roles under the hood, apart from "Good standing" which is a lot more complicated, and as such we have descoped for now. TL;DR all "base roles" other than "Good standing" will appear in the filter modal.

## Related Tickets & Documents



- Related Issue https://github.com/forem/forem/issues/17489
- Closes https://github.com/forem/forem/issues/17492

## QA Instructions, Screenshots, Recordings

- Make sure you have the `member_index_view` feature flag enabled and visit `/admin/member_manager/users`
- Open the filters modal
- Check the Status section shows the statuses (base roles) other than "Good standing"
- Check the behaviour selecting and clearing the checkboxes
- Try filtering by a status /several statuses 
- The applied status filters should also be visible at the top of the index page, and clicking X should remove them


https://user-images.githubusercontent.com/20773163/178496450-71691f30-dd53-4097-8656-ba9a58e09c00.mp4


### UI accessibility concerns?

Nothing new here

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [X] This change does not need to be communicated, and this is why not: comms will happen when feature flag is removed
